examples/training/semantic_segmentation/pascal_voc/fcn.py

```diff
@@ -0,0 +1,161 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Title: Train an Semantic Segmentation Model on Pascal VOC 2012 using KerasCV
+Author: [tanzhenyu](https://github.com/tanzhenyu)
+Date created: 2022/10/25
+Last modified: 2022/10/25
+Description: Use KerasCV to train a FCN on Pascal VOC 2012.
+"""
+
+import sys
+
+import tensorflow as tf
+from absl import flags
+from absl import logging
+
+from keras_cv.datasets.pascal_voc.segmentation import load
+from keras_cv.models.segmentation.fcn import FCN
+
+flags.DEFINE_string(
+    "weights_path",
+    "weights_{epoch:02d}.h5",
+    "Directory which will be used to store weight checkpoints.",
+)
+flags.DEFINE_boolean(
+    "mixed_precision",
+    True,
+    "Whether or not to use FP16 mixed precision for training.",
+)
+flags.DEFINE_string(
+    "tensorboard_path",
+    "logs",
+    "Directory which will be used to store tensorboard logs.",
+)
+FLAGS = flags.FLAGS
+FLAGS(sys.argv)
+
+if FLAGS.mixed_precision:
+    logging.info("mixed precision training enabled")
+    tf.keras.mixed_precision.set_global_policy("mixed_float16")
+
+# Try to detect an available TPU. If none is present, default to MirroredStrategy
+try:
+    tpu = tf.distribute.cluster_resolver.TPUClusterResolver.connect()
+    strategy = tf.distribute.TPUStrategy(tpu)
+except ValueError:
+    # MirroredStrategy is best for a single machine with one or multiple GPUs
+    strategy = tf.distribute.MirroredStrategy()
+print("Number of accelerators: ", strategy.num_replicas_in_sync)
+
+# parameters from FasterRCNN [paper](https://arxiv.org/pdf/1506.01497.pdf)
+
+local_batch = 4
+global_batch = local_batch * strategy.num_replicas_in_sync
+base_lr = 0.007 * global_batch / 16
+
+# TODO(tanzhenyu): add a diff dataset.
+# all_ds = load(split="sbd_train", data_dir=None)
+# all_ds = all_ds.concatenate(load(split="sbd_eval", data_dir=None))
+# train_ds = all_ds.take(10000)
+# eval_ds = all_ds.skip(10000).concatenate(load(split="diff", data_dir=None))
+train_ds = load(split="sbd_train", data_dir=None)
+eval_ds = load(split="sbd_eval", data_dir=None)
+
+resize_layer = tf.keras.layers.Resizing(512, 512, interpolation="nearest")
+
+image_size = [512, 512, 3]
+
+
+# TODO(tanzhenyu): move to KPL.
+def flip_fn(image, cls_seg):
```

Comment:
Try using KCV's preprocessing layer for flipping images

Reply:
Will use KCV's preprocessing layer !

Reply:
The discussion below talks about potentially making a general script instead of this one - a PR is already proposed that uses KPLs here and simplifies the preprocessing at #1183

Reply:
Yeah @DavidLandup0 we can remove this script and use the this #1183 once it gets merged!